Snoop Filter

gzh-terry opened the pull request.

huangyulong3 reviewed (approved):
LGTM.

chejinxian reviewed (approved):
LGTM

huangyulong3 commented on service/utils/btsnoop_filter.h:
```diff
@@ -61,27 +62,14 @@ typedef enum {
+    BTSNOOP_L2CAP_RSP_RESULT_PSM_NOT_SUPPORTED = 0x02,
+    BTSNOOP_L2CAP_RSP_RESULT_SECURITY_BLOCK = 0x03,
+    BTSNOOP_L2CAP_RSP_RESULT_NO_RESOURCES_AVAILABLE = 0x04,
+    BTSNOOP_L2CAP_RSP_RESULT_INVALID_SCID = 0x06,
```
Is there a specific reason why the enumeration values here jump from 4 directly to 6 without including 5?

gzh-terry replied:
Not sure, it was reserved before (v4.2) and is removed later
![image](https://github.com/user-attachments/assets/69ff5038-cf9b-4221-9279-286a3bb75746)

JinxianChe commented on service/utils/btsnoop_filter.c:
```diff
@@ -29,46 +31,72 @@ static g_snoop_filter_global_t g_snoop_filter = { 0 };
-            peer_cid = get_scid(pkt);                                                       \
-            response_cids_info.peer_cid = peer_cid;                                         \
-        }                                                                                   \
+#define GET_HCI_EVENT_CODE(hci_pkt) ((hci_pkt)[0])
```
Isn't this the same as the GET_HCI_TYPE macro here?
Should it be `#define GET_HCI_EVENT_CODE(hci_pkt) ((hci_pkt)[1])` here?

gzh-terry replied:
Yes, you are correct.
Should keep it as `evt`

gzh-terry replied:
Done

gzh-terry committed:
btsnoop: allow to remove RFCOMM data and NoCP events.

bug: v/61175

Rootcause: RFCOMM and NoCP events are too frequent
and may cause memory issues.

Signed-off-by: Zihao Gao <gaozihao@xiaomi.com>

gzh-terry committed:
SPP: fix memleak

bug: v/61175

Rootcause: the input argument `buffer` is supposed to be freed when `euv_pipe_write` completed. Memory leak may occurs when `device` does not exist but `buffer` is not null. Same when `euv_pipe_write` fails.

Signed-off-by: Zihao Gao <gaozihao@xiaomi.com>

huangyulong3 reviewed (approved):
LGTM.

huangyulong3 reviewed (approved):
LGTM.
